java/parser.js

```diff
@@ -1,171 +1,206 @@
 const types = require('./mapping/types')
 const selectors = require('./mapping/selectors')
-const {capitalizeFirstLetter} = require('./util')
+const {capitalizeFirstLetter, isEmpty} = require('./util')
+const {CHECK_SETTINGS_HOOKS, CHECK_SETTINGS_OPTIONS} = require('./mapping/supported')
+
 
 function checkSettings(cs, native) {
+    checkOptions(cs, CHECK_SETTINGS_OPTIONS)
     let java = `Target`
-    if (cs === undefined) {
+    if (cs === undefined || isEmpty(cs)) {
         return java + '.window()'
     }
     let element = ''
     let options = ''
     if (cs.frames === undefined && cs.region === undefined) element = '.window()'
     else {
-        if (cs.frames) element += frames(cs.frames)
+        if (cs.frames) {
+            if (cs.scrollRootElement) {
+                element += `.window().scrollRootElement(${printSelector(cs.scrollRootElement)})`
+            }
+            element += frames(cs.frames)
+        }
         if (cs.region) element += region(cs.region)
     }
-    if (cs.ignoreRegions) options += ignoreRegions(cs.ignoreRegions)
-    if (cs.floatingRegions) options += floatingRegions(cs.floatingRegions)
-    if (cs.accessibilityRegions) options += accessibilityRegions(cs.accessibilityRegions)
-    if (cs.layoutRegions) options += layoutRegions(cs.layoutRegions)
-    if (cs.scrollRootElement && !cs.frames) options += `.scrollRootElement(${printSelector(cs.scrollRootElement)})`
-    if (cs.ignoreDisplacements !== undefined) options += `.ignoreDisplacements(${cs.ignoreDisplacements})`
-    if (cs.sendDom !== undefined) options += `.sendDom(${serialize(cs.sendDom)})`
-    if (cs.matchLevel) options += `.matchLevel(MatchLevel.${cs.matchLevel.toUpperCase()})`
-    if (cs.name) options += `.withName(${cs.name})`
-    if (cs.layoutBreakpoints) options += `.layoutBreakpoints(${cs.layoutBreakpoints})`
+    if (cs.ignoreRegions) options += ignoreRegions(cs.ignoreRegions);
+    if (cs.floatingRegions) options += floatingRegions(cs.floatingRegions);
+    if (cs.accessibilityRegions) options += accessibilityRegions(cs.accessibilityRegions);
+    if (cs.layoutRegions) options += layoutRegions(cs.layoutRegions);
+    if (cs.scrollRootElement && !cs.frames) options += `.scrollRootElement(${printSelector(cs.scrollRootElement)})`;
+    if (cs.ignoreDisplacements !== undefined) options += `.ignoreDisplacements(${cs.ignoreDisplacements})`;
+    if (cs.timeout) options += `.timeout(${serialize(cs.timeout)})`;
+    if (cs.sendDom !== undefined) options += `.sendDom(${serialize(cs.sendDom)})`;
+    if (cs.matchLevel) options += `.matchLevel(MatchLevel.${cs.matchLevel.toUpperCase()})`;
+    if (cs.name) options += `.withName("${cs.name}")`;
+    if (cs.layoutBreakpoints) options += `.layoutBreakpoints(${cs.layoutBreakpoints})`;
     if (cs.isFully === true) {
-        options += '.fully()'
+        options += '.fully()';
     } else if (cs.isFully === false) {
-        options += '.fully(false)'
+        options += '.fully(false)';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/sdk.coverage.tests/pullRequest?prid=8528488)
```suggestion
        options += ".fully(false)";
```